lib/ex_doc/formatter/epub/templates/nav_grouped_modules_template.eex

```diff
@@ -0,0 +1,19 @@
+<% nodes = Map.new(nodes, fn %{id: id} = node -> {id, node} end) %>
+<li><span><%= name %></span>
+  <ol>
+  <%= for {title, modules} <- config.groups_for_modules, is_list(modules) do %>
```

Comment:
There are no guarantees that all the available modules are listed here. So, I guess we better put those modules at the end in another loop.

Reply:
I just noticed that based on the docs from `lib/mix/tasks/docs.ex`, `groups_for_modules` can also hold 

> A regex or the string name of the module is also supported.

So, I should adapt this template to support those cases. Or add an intermediate function that adapts the output and then delegates that to `nav_grouped_modules_template` to make it cleaner.

Reply:
The group for a module is already in the module struct and I believe they are already sorted according to groups. So I think a `Enum.chunk_by(modules, & &1.group)` may be enough to get groups and the correct order. Otherwise we can likely use the same algorithm/logic we use for extras.

Reply:
I realized we could also refactor extras with my suggestion above, so here it goes: 3fbe0bc344b7d30edb87460db518f786fb7354bf :)

Reply:
@josevalim Thanks for the context and advice, now the logic for extras and modules is the same at least in the table of content generation, here is the result:

<img width="934" alt="Screenshot 2024-01-24 at 7 39 07 p m" src="https://github.com/elixir-lang/ex_doc/assets/34700/69a93b36-faa7-46a8-9f54-fbf3a9c9360d">
<img width="885" alt="Screenshot 2024-01-24 at 7 38 56 p m" src="https://github.com/elixir-lang/ex_doc/assets/34700/2703c48a-2ef8-4c76-a674-74db6c84927f">